apps/web/pages/api/book/event.ts

```diff
@@ -377,11 +377,33 @@ async function handler(req: NextApiRequest) {
 
   const customInputs = {} as NonNullable<CalendarEvent["customInputs"]>;
 
+  const customInputsPrisma = await prisma.eventType.findUnique({
+    where: {
+      id: eventTypeId,
+    },
+    include: {
+      customInputs: {
+        where: {
+          required: true,
+        },
+      },
+    },
+  });
   if (reqBody.customInputs.length > 0) {
     reqBody.customInputs.forEach(({ label, value }) => {
       customInputs[label] = value;
     });
   }
+  const customInputEmpty: Array<string> = [];
+  customInputsPrisma?.customInputs.forEach(({ label }) => {
+    if (!customInputs[label]) {
+      customInputEmpty.push(label);
+    }
+  });
+
+  if (customInputEmpty.length > 0) {
+    throw new HttpError({ statusCode: 400, message: `Missing input ${customInputEmpty.toString()}` });
```

Comment:
I don't that's needed. With your changes in the BookingPage we already make sure that the form can't be submitted when required fields are empty

Reply:
Hi @CarinaWolli, thanks for the review! 

This was a suggestion from @zomars when we first pushed the validation
https://github.com/calcom/cal.com/pull/2829#pullrequestreview-985351190

what should we do next?

Reply:
You're both right @CarinaWolli and @gitstart the BookingPage prevents required inputs to be missing in the frontend. But in order for this to be a complete solution we need to validate in the backend as well so someone cannot bypass the call by modifying the request. 

This solution may work but I would rather try to use a dynamic zod schema validation for these use cases. This validation should happen as early as possible in the request.